crates/client/src/client.rs

```diff
@@ -464,7 +464,7 @@ impl FuelClient {
         self.query(query).await.map(|r| r.execute)
     }
 
-    pub async fn register(&self, id: &str, register: RegisterId) -> io::Result<Word> {
+    pub async fn register(&self, id: &str, register: u32) -> io::Result<Word> {
```

Comment:
Going from `RegisterId` to `u32` appears to be losing domain specificity. Can the `RegisterId` type alias `u32` instead?